### ISSUE TITLE
The code does not figure the figures of the research paper

### ISSUE THREAD

**Issue by BassantTolba1234** [YudiDong/Channel_Estimation_cGAN#6]
Please sir,
I tried to run your code to reproduce your paper figure of SNR Vs NMSE. But the generated curve does not reflect the curve in the paper !!
please can you kindly provide the correct code that reproduces correctly your research paper figures?
Thanks in advance.

**Comment by xiaoningaixuexi** [comment]
Please sir ,
Have you solved this problem? I have encountered the same problem, so if you have a solution to it, please send me a copy, I will be very grateful.

**Comment by wyjend** [comment]
俺也一样，有复现成功的hxd吗

**Comment by jameslin1504** [comment]
Sir
Can you provide the cGAN model  in github? I want to compare the performance  of different generative models.

**Comment by zzjng** [comment]
代码提供的网络结构好像跟论文中图4的结构不太一样

**Comment by xuelianp** [comment]
> 代码提供的网络结构好像跟论文中图4的结构不太一样

The loss function calculation in the code seems to be inconsistent with the loss function calculation in the paper, is this the same problem you found?

**Comment by zzjng** [comment]
> > 代码提供的网络结构好像跟论文中图4的结构不太一样
> 
> The loss function calculation in the code seems to be inconsistent with the loss function calculation in the paper, is this the same problem you found?

No, I mean the structure of the generator in the code is inconsistent with it in the paper.

**Comment by zzjng** [comment]
> 俺也一样，有复现成功的hxd吗

snr_nmse d的图出不来，用代码跑出来的NMSE-epoch图 数据没有论文里的那么好，论文里面大概0.02左右，我跑出来的最低0.2多，再试试吧。博主提供的能跑就很好了，或者说能提供就很好了

**Comment by xuelianp** [comment]
Thank you very much for your reply, maybe I didn't express myself clearly before, I mean the same thing as you.


---Original---
From: ***@***.***&gt;
Date: Mon, May 29, 2023 20:40 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [YudiDong/Channel_Estimation_cGAN] The code does not figure thefigures of the research paper (Issue #6)




   
代码提供的网络结构好像跟论文中图4的结构不太一样
  
The loss function calculation in the code seems to be inconsistent with the loss function calculation in the paper, is this the same problem you found?
  
No, I mean the structure of the generator in the code is inconsistent with it in the paper.

**Comment by xuelianp** [comment]
我也没复现成功，正在尝试中，我还没弄snr_nmse的图，数据没有论文里面的好，差了有十倍，可能是py那边的网络参数改了，你可以尝试修改一下。不是博主，尝试复现工作而已。如果你能提供一些相关代码给我就好了&nbsp;





&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年5月29日(星期一) 晚上8:46
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [YudiDong/Channel_Estimation_cGAN] The code does not figure the figures of the research paper (Issue #6)





  
俺也一样，有复现成功的hxd吗
  
snr_nmse d的图出不来，用代码跑出来的NMSE-epoch图 数据没有论文里的那么好，论文里面大概0.02左右，我跑出来的最低0.2多，再试试吧。博主提供的能跑就很好了，或者说能提供就很好了
